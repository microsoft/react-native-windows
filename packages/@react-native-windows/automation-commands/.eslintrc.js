module.exports = {
    extends: ['@rnw-scripts'],
    parserOptions: {tsconfigRootDir : __dirname},
};
