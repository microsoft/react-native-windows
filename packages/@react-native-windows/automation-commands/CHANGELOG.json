{
  "name": "@react-native-windows/automation-commands",
  "entries": [
    {
      "date": "Fri, 01 Oct 2021 05:05:54 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.6-2",
      "version": "0.0.6-2",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.5-2",
            "author": "julio.rocha@microsoft.com",
            "commit": "eaefea114554d0b91dbf2d6aff650b2fc5673f7a",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 30 Sep 2021 05:06:48 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.6-1",
      "version": "0.0.6-1",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.5-1",
            "author": "erozell@outlook.com",
            "commit": "4bc8637395ea828788c0ba267561c1a01f06f484",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 29 Sep 2021 05:06:54 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.6-0",
      "version": "0.0.6-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.5-0",
            "author": "anandrag@microsoft.com",
            "commit": "eb04aeb2e5fcb74988f1630f24adc27a034351fe",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Tue, 28 Sep 2021 05:07:03 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.5",
      "version": "0.0.5",
      "comments": {
        "patch": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.4",
            "author": "34109996+chiaramooney@users.noreply.github.com",
            "commit": "d4970fdd583866c3ac7af6a0cf38040cf3b22d93",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 25 Sep 2021 05:06:39 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.5-0",
      "version": "0.0.5-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-12",
            "author": "53799235+ZihanChen-MSFT@users.noreply.github.com",
            "commit": "eaf932469678b8b67b0b2b17fd2bcdda1ed4e358",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 24 Sep 2021 05:06:42 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.4",
      "version": "0.0.4",
      "comments": {
        "patch": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-11",
            "author": "igklemen@microsoft.com",
            "commit": "8b63ada9ba1d55bfd161388fc18c2b5c6445b508",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 23 Sep 2021 05:06:48 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.4-1",
      "version": "0.0.4-1",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-10",
            "author": "34109996+chiaramooney@users.noreply.github.com",
            "commit": "3813907b2eaa6198f0a8b9dc570075f0c7117301",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 22 Sep 2021 05:06:43 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.4-0",
      "version": "0.0.4-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-9",
            "author": "ali-hk@users.noreply.github.com",
            "commit": "0bc36a30ab3cd5bb4cfd767e503b29c185cfcddd",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Tue, 21 Sep 2021 05:08:39 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3",
      "version": "0.0.3",
      "comments": {
        "patch": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-8",
            "author": "igklemen@microsoft.com",
            "commit": "08eb19f42f4f74bd82f33db2e27ba2d33b2ddd1b",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 18 Sep 2021 05:07:45 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-7",
      "version": "0.0.3-7",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-7",
            "author": "erozell@outlook.com",
            "commit": "fbbfb83843c60abb8c02368077fd0a406f0bdcc0",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 17 Sep 2021 05:06:53 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-6",
      "version": "0.0.3-6",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-6",
            "author": "jthysell@microsoft.com",
            "commit": "95fecc02d3dcc9f03d0fc7d47d30cb45e128bca6",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 16 Sep 2021 05:06:44 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-5",
      "version": "0.0.3-5",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-5",
            "author": "erozell@outlook.com",
            "commit": "b46965a75f179c1855a2499c7f3c6b4b10daf416",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 15 Sep 2021 05:07:04 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-4",
      "version": "0.0.3-4",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-4",
            "author": "30809111+acoates-ms@users.noreply.github.com",
            "commit": "5dbd74b1852c38f851483478d6da7d81949d0195",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Tue, 14 Sep 2021 05:07:02 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-3",
      "version": "0.0.3-3",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-3",
            "author": "jthysell@microsoft.com",
            "commit": "d3de4a2b04415ff9004b7ba08ee21787b924d0d4",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 11 Sep 2021 05:06:55 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-2",
      "version": "0.0.3-2",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-2",
            "author": "anandrag@microsoft.com",
            "commit": "09bc9974c8721bf5559680dc7a8aa00514c1c7f3",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 10 Sep 2021 05:07:31 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-1",
      "version": "0.0.3-1",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-1",
            "author": "53799235+ZihanChen-MSFT@users.noreply.github.com",
            "commit": "42d0853c684f0e9f9c9f1f7fa34cbf4ca3810edc",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Thu, 09 Sep 2021 05:07:46 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.3-0",
      "version": "0.0.3-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.3-0",
            "author": "53799235+ZihanChen-MSFT@users.noreply.github.com",
            "commit": "66a9f86f24f862c9e9e611ad435f68daddb4e04e",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Wed, 08 Sep 2021 05:08:53 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.2",
      "version": "0.0.2",
      "comments": {
        "patch": [
          {
            "comment": "Set consistent node requirements on our packages",
            "author": "ngerlem@microsoft.com",
            "commit": "2974ea0ab58b546264b8d9a4a7c12ceeb0a02851",
            "package": "@react-native-windows/automation-commands"
          }
        ],
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.2",
            "author": "ngerlem@microsoft.com",
            "commit": "2974ea0ab58b546264b8d9a4a7c12ceeb0a02851",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Sat, 04 Sep 2021 05:06:39 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.2-0",
      "version": "0.0.2-0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.2-0",
            "author": "dannyvv@microsoft.com",
            "commit": "d0c3465f27bd0de739b23b46e957cbdab6bdf014",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    },
    {
      "date": "Fri, 03 Sep 2021 05:06:04 GMT",
      "tag": "@react-native-windows/automation-commands_v0.0.1",
      "version": "0.0.1",
      "comments": {
        "patch": [
          {
            "comment": "@react-native-windows/automation",
            "author": "ngerlem@microsoft.com",
            "commit": "17e32d36180fa5c7e45c3c926c6eaa729efb6ea0",
            "package": "@react-native-windows/automation-commands"
          }
        ],
        "prerelease": [
          {
            "comment": "Bump @react-native-windows/automation-channel to v0.0.1",
            "author": "erozell@outlook.com",
            "commit": "439b8a54b62807068bfa17e130f8ab0378eaaf36",
            "package": "@react-native-windows/automation-commands"
          }
        ]
      }
    }
  ]
}
