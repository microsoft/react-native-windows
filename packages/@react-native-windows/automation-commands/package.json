{
  "name": "@react-native-windows/automation-commands",
  "version": "0.1.31",
  "description": "Allows controlling your react-native-windows application",
  "main": "lib-commonjs/index.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git@github.com:microsoft/react-native-windows.git",
    "directory": "packages/@react-native-windows/automation-commands"
  },
  "private": true,
  "scripts": {
    "build": "rnw-scripts build",
    "clean": "rnw-scripts clean",
    "lint": "rnw-scripts lint",
    "lint:fix": "rnw-scripts lint:fix",
    "watch": "rnw-scripts watch"
  },
  "dependencies": {
    "@react-native-windows/automation-channel": "^0.5.11",
    "@typescript-eslint/eslint-plugin": "^5.30.5",
    "@typescript-eslint/parser": "^5.30.5"
  },
  "devDependencies": {
    "@jest/types": "^29.2.1",
    "@rnw-scripts/eslint-config": "1.1.14",
    "@rnw-scripts/just-task": "2.3.3",
    "@rnw-scripts/ts-config": "2.0.2",
    "@types/jest": "^29.2.2",
    "@types/node": "^14.14.22",
    "eslint": "^8.19.0",
    "prettier": "^2.4.1",
    "typescript": "^4.9.5"
  },
  "files": [
    "lib-commonjs",
    "README.md"
  ],
  "engines": {
    "node": ">= 14"
  }
}