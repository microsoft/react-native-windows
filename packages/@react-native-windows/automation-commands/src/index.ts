/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import dumpVisualTree from './dumpVisualTree';

export {dumpVisualTree};
