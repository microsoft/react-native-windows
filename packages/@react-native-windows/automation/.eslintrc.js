module.exports = {
  extends: ['@rnw-scripts'],
  parserOptions: {tsconfigRootDir : __dirname},
  
  // Workaround "export type" being unsupported with prettier 1.x
  ignorePatterns: ['index.ts'],
  overrides: [
    {
      files: ['*.ts'],
      // Also not clean
      rules: {
        '@typescript-eslint/no-unused-vars': 'off',
      },
    },
  ],
};
