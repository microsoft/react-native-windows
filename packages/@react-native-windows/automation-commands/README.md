# @react-native-windows/automation-commands

`@react-native-windows/automation-commands` provides built-in commands that may
be used with `@react-native-windows/automation-channel` to make it easier to
test your application.

**This package is a work in progress**

## Adding to your application

**TBD**

## Commands

### dumpVisualTree

The `dumpVisualTree` command creates a JSON object corresponding to the XAML
tree under a given testID. This can be used in conjunction with snapshot
testing to validate that your native UI looks how you could expect.

```ts
import {dumpVisualTree} from '@react-native-windows/automation-commands';

test('Widget', async () => {
  const dump = await dumpVisualTree('widget-test-id');
  expect(dump).toMatchSnapshot();
});
```

#### Options
**TBD**