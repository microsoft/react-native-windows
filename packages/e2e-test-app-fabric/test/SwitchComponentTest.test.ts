/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import {dumpVisualTree} from '@react-native-windows/automation-commands';
import {goToComponentExample} from './RNTesterNavigation';
import {verifyNoErrorLogs} from './Helpers';
import {app} from '@react-native-windows/automation';

const searchBox = async (input: string) => {
  const searchBox = await app.findElementByTestID('example_search');
  await app.waitUntil(
    async () => {
      await searchBox.setValue(input);
      return (await searchBox.getText()) === input;
    },
    {
      interval: 1500,
      timeout: 5000,
      timeoutMsg: `Unable to enter correct search text into test searchbox.`,
    },
  );
};

beforeAll(async () => {
  // If window is partially offscreen, tests will fail to click on certain elements
  await app.setWindowPosition(0, 0);
  await app.setWindowSize(1000, 1250);
  await goToComponentExample('Switch');
});

afterEach(async () => {
  await verifyNoErrorLogs();
});

describe('SwitchTests', () => {
  test('Switches can be set to true/false, initial false', async () => {
    await searchBox('tru');
    const component = await app.findElementByTestID('on-off-initial-off');
    await component.waitForDisplayed({timeout: 5000});
    const dump = await dumpVisualTree('on-off-initial-off');
    expect(dump).toMatchSnapshot();
    await component.click();
    const dump2 = await dumpVisualTree('on-off-initial-off');
    expect(dump2).toMatchSnapshot();
  });

  test('Switches can be set to true/false, initial true', async () => {
    await searchBox('tru');
    const dump = await dumpVisualTree('on-off-initial-on');
    expect(dump).toMatchSnapshot();
  });

  test('Switches can be disabled, initial false', async () => {
    await searchBox('dis');
    const component = await app.findElementByTestID('disabled-initial-off');
    await component.waitForDisplayed({timeout: 5000});
    const dump = await dumpVisualTree('disabled-initial-off');
    expect(dump).toMatchSnapshot();
    await component.click();
    const dump2 = await dumpVisualTree('disabled-initial-off');
    expect(dump2).toMatchSnapshot();
  });

  test('Switches can be disabled, initial true', async () => {
    await searchBox('dis');
    const dump = await dumpVisualTree('disabled-initial-on');
    expect(dump).toMatchSnapshot();
  });

  test('Custom colors can be provided, initial false', async () => {
    await searchBox('cus');
    const dump = await dumpVisualTree('initial-false-switch');
    expect(dump).toMatchSnapshot();
  });

  test('Custom colors can be provided, initial true', async () => {
    await searchBox('cus');
    const dump = await dumpVisualTree('initial-true-switch');
    expect(dump).toMatchSnapshot();
  });

  test('Change events can be detected, event top', async () => {
    await searchBox('cha');
    const dump = await dumpVisualTree('event-switch-top');
    expect(dump).toMatchSnapshot();
  });

  test('Change events can be detected, event bottom', async () => {
    await searchBox('cha');
    const dump = await dumpVisualTree('event-switch-bottom');
    expect(dump).toMatchSnapshot();
  });

  test('Change events can be detected, event regression top', async () => {
    await searchBox('cha');
    const dump = await dumpVisualTree('event-switch-regression-top');
    expect(dump).toMatchSnapshot();
  });

  test('Change events can be detected, events regression bottom', async () => {
    await searchBox('cha');
    const dump = await dumpVisualTree('event-switch-regression-bottom');
    expect(dump).toMatchSnapshot();
  });
  test('Controlled switch', async () => {
    await searchBox('con');
    const dump = await dumpVisualTree('controlled-switch');
    expect(dump).toMatchSnapshot();
  });
  test('Switches can have customized accessibility', async () => {
    await searchBox('acc');
    const dump = await dumpVisualTree('accessibility-switch');
    expect(dump).toMatchSnapshot();
  });
});
