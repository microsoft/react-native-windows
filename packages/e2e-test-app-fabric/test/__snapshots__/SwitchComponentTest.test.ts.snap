// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchTests Change events can be detected, event bottom 1`] = `
{
  "Automation Tree": {
    "AutomationId": "event-switch-bottom",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "event-switch-bottom",
    },
  },
  "Visual Tree": {
    "Comment": "event-switch-bottom",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "4, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Change events can be detected, event regression top 1`] = `
{
  "Automation Tree": {
    "AutomationId": "event-switch-regression-top",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "event-switch-regression-top",
    },
  },
  "Visual Tree": {
    "Comment": "event-switch-regression-top",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "24, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Change events can be detected, event top 1`] = `
{
  "Automation Tree": {
    "AutomationId": "event-switch-top",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "event-switch-top",
    },
  },
  "Visual Tree": {
    "Comment": "event-switch-top",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "4, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Change events can be detected, events regression bottom 1`] = `
{
  "Automation Tree": {
    "AutomationId": "event-switch-regression-bottom",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "event-switch-regression-bottom",
    },
  },
  "Visual Tree": {
    "Comment": "event-switch-regression-bottom",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "24, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Controlled switch 1`] = `
{
  "Automation Tree": {
    "AutomationId": "controlled-switch",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "controlled-switch",
    },
  },
  "Visual Tree": {
    "Comment": "controlled-switch",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "4, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Custom colors can be provided, initial false 1`] = `
{
  "Automation Tree": {
    "AutomationId": "initial-false-switch",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "initial-false-switch",
    },
  },
  "Visual Tree": {
    "Comment": "initial-false-switch",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "4, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Custom colors can be provided, initial true 1`] = `
{
  "Automation Tree": {
    "AutomationId": "initial-true-switch",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "initial-true-switch",
    },
  },
  "Visual Tree": {
    "Comment": "initial-true-switch",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "24, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Switches can be disabled, initial false 1`] = `
{
  "Automation Tree": {
    "AutomationId": "disabled-initial-off",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "disabled-initial-off",
    },
  },
  "Visual Tree": {
    "Comment": "disabled-initial-off",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "4, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Switches can be disabled, initial true 1`] = `
{
  "Automation Tree": {
    "AutomationId": "disabled-initial-on",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "disabled-initial-on",
    },
  },
  "Visual Tree": {
    "Comment": "disabled-initial-on",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "24, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Switches can be set to true/false, initial false 1`] = `
{
  "Automation Tree": {
    "AutomationId": "on-off-initial-off",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "on-off-initial-off",
    },
  },
  "Visual Tree": {
    "Comment": "on-off-initial-off",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "4, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Switches can be set to true/false, initial true 1`] = `
{
  "Automation Tree": {
    "AutomationId": "on-off-initial-on",
    "ControlType": 50000,
    "IsKeyboardFocusable": true,
    "LocalizedControlType": "button",
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
    "_Props": {
      "TestId": "on-off-initial-on",
    },
  },
  "Visual Tree": {
    "Comment": "on-off-initial-on",
    "Offset": "0, 0, 0",
    "Size": "51, 31",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "5.5, 5.5, 0",
        "Size": "40, 20",
        "Visual Type": "Visual",
        "__Children": [
          {
            "Offset": "24, 4, 0",
            "Size": "12, 12",
            "Visual Type": "Visual",
          },
        ],
      },
    ],
  },
}
`;

exports[`SwitchTests Switches can have customized accessibility 1`] = `
{
  "Automation Tree": {
    "AutomationId": "accessibility-switch",
    "ControlType": 50026,
    "LocalizedControlType": "group",
    "__Children": [
      {
        "AutomationId": "",
        "ControlType": 50000,
        "HelpText": "Switch",
        "IsKeyboardFocusable": true,
        "LocalizedControlType": "button",
        "Name": "This is a Switch",
      },
      {
        "AutomationId": "",
        "ControlType": 50000,
        "HelpText": "Switch",
        "IsKeyboardFocusable": true,
        "LocalizedControlType": "button",
        "Name": "This is a Switch with focusable set to false",
      },
      {
        "AutomationId": "",
        "ControlType": 50000,
        "HelpText": "Switch",
        "IsKeyboardFocusable": true,
        "LocalizedControlType": "button",
        "Name": "This is a Switch with accessible set to false",
      },
      {
        "AutomationId": "",
        "ControlType": 50000,
        "HelpText": "Switch",
        "IsKeyboardFocusable": true,
        "LocalizedControlType": "button",
        "Name": "This is a Switch with focusable and accessible set to false",
      },
    ],
  },
  "Component Tree": {
    "Type": "Microsoft.ReactNative.Composition.ViewComponentView",
    "_Props": {
      "TestId": "accessibility-switch",
    },
    "__Children": [
      {
        "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
        "_Props": {
          "AccessibilityLabel": "This is a Switch",
        },
      },
      {
        "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
        "_Props": {
          "AccessibilityLabel": "This is a Switch with focusable set to false",
        },
      },
      {
        "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
        "_Props": {
          "AccessibilityLabel": "This is a Switch with accessible set to false",
        },
      },
      {
        "Type": "Microsoft.ReactNative.Composition.SwitchComponentView",
        "_Props": {
          "AccessibilityLabel": "This is a Switch with focusable and accessible set to false",
        },
      },
    ],
  },
  "Visual Tree": {
    "Comment": "accessibility-switch",
    "Offset": "0, 0, 0",
    "Size": "916, 82",
    "Visual Type": "SpriteVisual",
    "__Children": [
      {
        "Offset": "0, 0, 0",
        "Size": "51, 31",
        "Visual Type": "SpriteVisual",
        "__Children": [
          {
            "Offset": "0, 0, 0",
            "Size": "51, 31",
            "Visual Type": "SpriteVisual",
            "__Children": [
              {
                "Offset": "5.5, 5.5, 0",
                "Size": "40, 20",
                "Visual Type": "Visual",
                "__Children": [
                  {
                    "Offset": "24, 4, 0",
                    "Size": "12, 12",
                    "Visual Type": "Visual",
                  },
                ],
              },
            ],
          },
          {
            "Offset": "0, 0, 0",
            "Size": "0, 0",
            "Visual Type": "SpriteVisual",
          },
        ],
      },
      {
        "Offset": "865, 0, 0",
        "Size": "51, 31",
        "Visual Type": "SpriteVisual",
        "__Children": [
          {
            "Offset": "0, 0, 0",
            "Size": "51, 31",
            "Visual Type": "SpriteVisual",
            "__Children": [
              {
                "Offset": "5.5, 5.5, 0",
                "Size": "40, 20",
                "Visual Type": "Visual",
                "__Children": [
                  {
                    "Offset": "24, 4, 0",
                    "Size": "12, 12",
                    "Visual Type": "Visual",
                  },
                ],
              },
            ],
          },
          {
            "Offset": "0, 0, 0",
            "Size": "0, 0",
            "Visual Type": "SpriteVisual",
          },
        ],
      },
      {
        "Offset": "0, 41, 0",
        "Size": "51, 31",
        "Visual Type": "SpriteVisual",
        "__Children": [
          {
            "Offset": "0, 0, 0",
            "Size": "51, 31",
            "Visual Type": "SpriteVisual",
            "__Children": [
              {
                "Offset": "5.5, 5.5, 0",
                "Size": "40, 20",
                "Visual Type": "Visual",
                "__Children": [
                  {
                    "Offset": "4, 4, 0",
                    "Size": "12, 12",
                    "Visual Type": "Visual",
                  },
                ],
              },
            ],
          },
          {
            "Offset": "0, 0, 0",
            "Size": "0, 0",
            "Visual Type": "SpriteVisual",
          },
        ],
      },
      {
        "Offset": "865, 41, 0",
        "Size": "51, 31",
        "Visual Type": "SpriteVisual",
        "__Children": [
          {
            "Offset": "0, 0, 0",
            "Size": "51, 31",
            "Visual Type": "SpriteVisual",
            "__Children": [
              {
                "Offset": "5.5, 5.5, 0",
                "Size": "40, 20",
                "Visual Type": "Visual",
                "__Children": [
                  {
                    "Offset": "24, 4, 0",
                    "Size": "12, 12",
                    "Visual Type": "Visual",
                  },
                ],
              },
            ],
          },
          {
            "Offset": "0, 0, 0",
            "Size": "0, 0",
            "Visual Type": "SpriteVisual",
          },
        ],
      },
    ],
  },
}
`;
